checker/src/org/checkerframework/checker/index/samelen/SameLenAnnotatedTypeFactory.java

```diff
@@ -108,7 +114,30 @@ public AnnotationMirror createCombinedSameLen(
             bValues.addAll(SameLenUtils.getValue(sl2));
         }
 
-        return getCombinedSameLen(aValues, bValues);
+        AnnotationMirror res = getCombinedSameLen(aValues, bValues);
+        return res;
+    }
+
+    public FlowExpressions.Receiver arrayReceiverFromString(
```

Comment:
In the offset branch, which I actually just merged, I moved that code to the GenericATF. So you could move this method there, rename it to some more generic, add documentation, and have getAnnotationFromJavaExpressionString call the new method.  Or can do it after this PR is merged.